execution.go

```diff
@@ -126,7 +128,8 @@ func (q *queryExecution) executeRootStep(step *QueryPlanStep) error {
 
 func (q *queryExecution) executeDocument(document string, serviceURL string, response interface{}) error {
 	req := NewRequest(document).
-		WithHeaders(GetOutgoingRequestHeadersFromContext(q.ctx))
+		WithHeaders(GetOutgoingRequestHeadersFromContext(q.ctx)).
+		WithOperationName(q.operationName)
```

Comment:
I think you'll need to add the operation name to the generated query document too, otherwise the downstream queries will fail as it won't be able to find the operation in the query?

Reply:
I think in the `buildBoundaryQueryDocuments` and `executeRootStep` functions

Reply:
I've added the `operationName` to the required documents. I haven't added any automated tests but have checked it fixes the issue for the example setup.